22.md

```diff
@@ -0,0 +1,189 @@
+    ```
+
+7. `WALLET` then pays the return bolt11 invoice and if suitable, send `sourceExchangeData` back to the `callback`
+8. `WALLET` waits to receive webhooks to confirm final destination settlement.
+
+
+### Exchange data format
+Exchange data can be provided by both parties during a transaction, `destExchangeData` and `sourceExchangeData`.
+
+```Typescript
+    "destExchangeData": {
+        "id": string, // The identifier for the transaction.
+        "currency": string, // The currency (ISO 4217) passed as query, as confirmation that it is accepted.
+        "price": number, // The sell price promised for the received Bitcoin in given currency.
+        "expiry": number, // The unix timestamp of when the fx rate will expire.
+        "pr": string // (Optional) Bolt11 invoice for the transaction, optional depending on whether an amount is passed across in the initial request.
+    }
+```
+
+```Typescript
+    "sourceExchangeData": {
+        "id": string, // The identifier for the transaction
+        "currency": string, // The currency (ISO 4217) the Bitcoin was bought with.
+        "price": number, // The price paid to buy the Bitcoin in the source currency.
+        "webhook": string // The webhook endpoint to receive updates on destination settlement.
+    }
+```
+
+### Supported payout options format
+```Typescript
+    "supportPayoutOptions": {
+        [{"id": string, // The identifier for the payout option
+        "name": string, // The name of the payout option
+        "logourl": string, // URL to an image of the payout logo for display purposes
+        "requiredFields": [ // Required fields object should be returned for each payout option with the fields required to process a transaction, these can vary depending on service
+            { "name": "accountName", "datatype": "string" },
+            { "name": "accountNumber", "datatype": "string" },
+            { "name": "mobileNumber", "datatype": "string" },
+            { "name": "branchCode", "datatype": "string" }
+        ]
+        }]
+    }
+```
+
+### Webhook format
+Once a transaction has either settled or failed at the destination payment provider, a webhook should be sent to the `webhook`.
+
+Request header \
+`Authorization: Bearer <authKey>`
+
+Request body
+```Typescript
+    {
+        "id": string, // The identifier for the transaction
+        "paymentStatus": string, // The status of the bolt11 invoice payment (`pending`, `success`, or `failed`)
+        "payoutStatus": string, // The status of the destination settlement (`pending`, `success`, or `failed`)
+        "refund": string, // (optional) The endpoint that should be used to request a refund if the settlement has failed
+    }
+```
+
+### Refunds
+If settlement has failed at the destination, a webhook will be received with status `failed`. Upon receipt of this webhook type, a service should `POST` to the `refund` with a valid bolt11 lightning invoice, which then should be paid by the `SERVICE`
```

Comment:
> a service should `POST` to the `refund`

This should be the wallet?

Reply:
Could just be the previous service in case of a multi hop tx.

Reply:
I think it's a typo. In this context `WALLET` paid the `SERVICE` but the payment failed to settle at the destination, so `WALLET` should `POST` to the `refund` with a valid bolt11 lightning invoice (to be paid by `SERVICE`)

Right?

Reply:
What I meant is the case where the payment flow is something like: